sample/scripts/updateSampleCode.ts

```diff
@@ -0,0 +1,165 @@
+    let node = document.createElement('h2');
+    node.innerHTML = 'JS Sample Code';
+    jsSampleCode.appendChild(node);
+    let startString = assembleRequireString(plugins, defaultFormat);
+    let pluginsString = assemblePluginsString(plugins);
+    let formatString = assembleFormatString(defaultFormat);
+    let optionsString = assembleOptionsString(pluginsString, formatString);
+    let endString = assembleEndString(optionsString);
+
+    let jsCodeString = startString.concat(pluginsString).concat(formatString).concat(optionsString).concat(endString);
+    jsSampleCode.appendChild(createElementFromContent('jsSampleCode', jsCodeString));
+}
+
+// Create an 'xmp' block so the code can be displayed as is
+function createElementFromContent(id: string, content: string): HTMLElement {
+    let node = document.createElement('xmp');
+    node.id = id;
+    document.body.insertBefore(node, document.body.childNodes[0]);
+    node.innerHTML = content;
+
+    return node;
+}
+
+function assemblePluginsString(plugins: EditorPlugin[]): string {
+    if (plugins) {
+        let pluginsString = 'var plugins = [\n';
+        plugins.forEach(plugin => {
+            if (plugin instanceof Watermark) {
+                pluginsString = pluginsString.concat('  new roosterPlugins.Watermark(),\n');
+            } else if (plugin instanceof ImageResizePlugin) {
+                pluginsString = pluginsString.concat('  new roosterPlugins.ImageResizePlugin(),\n');
+            } else if (plugin instanceof DefaultShortcut) {
+                pluginsString = pluginsString.concat('  new roosterPlugins.DefaultShortcut(),\n');
+            } else if (plugin instanceof HyperLink) {
+                pluginsString = pluginsString.concat('  new roosterPlugins.HyperLink(),\n');
+            } else if (plugin instanceof PasteManager) {
+                pluginsString = pluginsString.concat('  new roosterPlugins.PasteManager(),\n');
+            } else if (plugin instanceof ContentEdit) {
+                pluginsString = pluginsString.concat('  new roosterPlugins.ContentEdit(),\n');
+            }
+        });
+        pluginsString = pluginsString.concat('];\n');
+
+        if (pluginsString != defaultPluginsString) {
+            return pluginsString;
+        }
+    }
+
+    return '';
+}
+
+function assembleFormatString(defaultFormat: DefaultFormat): string {
+    if (!defaultFormat) {
+        return '';
+    }
+    let defaultFormatString = 'var defaultFormat = {};\n'
+    if (defaultFormat.bold) {
+        defaultFormatString = defaultFormatString.concat('defaultFormat.bold = true;\n');
+    }
+    if (defaultFormat.italic) {
+        defaultFormatString = defaultFormatString.concat('defaultFormat.italic = true;\n');
+    }
+    if (defaultFormat.underline) {
+        defaultFormatString = defaultFormatString.concat('defaultFormat.underline = true;\n');
+    }
+    defaultFormatString = defaultFormatString.concat(`defaultFormat.textColor = ${defaultFormat.textColor};\n`);
```

Comment:
You need '' around textColor and fontFamily

Reply:
Added